src/middleware/index.ts

```diff
@@ -139,14 +140,37 @@ export const validateProvider = async (req: Request, _res: Response, next: NextF
         `Provider [${providerId}] not found or does not have any visible cloud hosted collections.`
       )
     );
-  } else if (!provider) {
+    // If it's not the 'ALL' provider and the provider ID cannot be found then throw an error
+  } else if (!provider && providerId != ALL_PROVIDER.toString()) {
     next(new ItemNotFound(`Provider [${providerId}] not found.`));
   } else {
     req.provider = provider;
     next();
   }
 };
 
+/**
+ * Middleware validates the provider in the route is not ALL.
+ *  
+ * This validation is only required for routes that will search CMR based on the provider
+ * supplied.
+ *
+ * If the provider is not 'ALL' then validation passes.
+ * If the provider is 'ALL', it exits early with a 404.
+
+ */
+export const validateNotAllProvider = async (req: Request, _res: Response, next: NextFunction) => {
```

Comment:
We could combine the logic in here within the existing `validateProvider` and just add another branch for the else conditional `if (providerId == ALL_PROVIDER.toString()) {
    next(
      new ItemNotFound(`This operation is not allowed for the ${ALL_PROVIDER.toString()} Catalog.`)
    );
  }`
  but, I also see the validity in keeping it separate because otherwise it does get a bit unruly. That said it does mean we are passing two very similar validations across the routes

Reply:
ValidateProvider applies to all routes (except two health routes). ValidateNotAllProvider only applies to some (we do not allow search routes and item routes for the ALL provider) so I don't think we can combine. . This is why I separated them. I think validateProvider is not necessary (see https://bugs.earthdata.nasa.gov/browse/CMR-10264) so we would probably remove that in the future. validateProvider checks to see if the provider exists via a call to CMR and then, if it does, adds it to the request. This decreases the efficiency of a valid call. An invalid provider should be caught at the graphQL call instead imho.